tests/ashley/api/test_api_manage_moderators.py

```diff
@@ -211,13 +212,184 @@ def test_access_can_manage_moderators_list_non_moderators(self):
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+        data = {"id": 10, "public_username": "Sab"}
+        response = self.client.put(
+            f"/api/v1.0/users/{user.id}/",
+            json.dumps(data),
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {"id": 1, "public_username": "user"})
+
+    def test_access_api_moderator_group_anonymous_patch(self):
+        """anonymous can't patch user to add or remove the group moderator."""
+        user = UserFactory(public_username="user")
+        action = random.choice(["add_group_moderator", "remove_group_moderator"])
+        response = self.client.patch(f"/api/v1.0/users/{user.id}/{action}/")
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response.json(), {"detail": "Authentication credentials were not provided."}
+        )
+
+    def test_access_api_moderator_group_student_patch(self):
+        """Student can't patch user to add or remove the group moderator."""
+        user = UserFactory(public_username="user")
+        # Creates a forum
+        lti_context = LTIContextFactory(lti_consumer=user.lti_consumer)
+        forum = ForumFactory()
+        forum.lti_contexts.add(lti_context)
+        assign_perm("can_read_forum", user, forum, True)
+
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+
+        action = random.choice(["add_group_moderator", "remove_group_moderator"])
+        response = self.client.patch(f"/api/v1.0/users/{user.id}/{action}/")
+        self.assertEqual(response.status_code, 403)
+        self.assertEqual(
+            response.json(),
+            {"detail": "You do not have permission to perform this action."},
+        )
+
+    def test_access_api_moderator_group_instructor_post(self):
+        """Instructor can't update user with a POST to add or remove the group moderator."""
+        user = UserFactory(public_username="user")
+        # Creates a forum
+        lti_context = LTIContextFactory(lti_consumer=user.lti_consumer)
+        forum = ForumFactory()
+        forum.lti_contexts.add(lti_context)
+        assign_perm("can_manage_moderator", user, forum, True)
+
+        # Creates the session
+        self.client.force_login(user, "ashley.auth.backend.LTIBackend")
+        session = self.client.session
+        session[SESSION_LTI_CONTEXT_ID] = lti_context.id
+        session.save()
+
+        action = random.choice(["add_group_moderator", "remove_group_moderator"])
+        response = self.client.post(f"/api/v1.0/users/{user.id}/{action}/")
```

Comment:
Note that POST should be on the list endpoint. That's why you get a 405. 
But ok you may want to confirm that this endpoint does not accept POST requests.